### ISSUE TITLE
About the validation splitting and evaluation

### ISSUE THREAD

**Issue by danielemalitesta** [jeongwhanchoi/BSPM#4]
Dear @jeongwhanchoi,

Thanks a lot for this amazing work, and for publishing the code on GitHub!

Me along other colleagues are working on your diffusion model for our research. We would like to run additional experiments on other datasets than those you used in your paper.

To do that, I assume we might need to run some kind of evaluation on the validation set, to find the best hyper-parameter values of the model. However, to the best of our understanding, we cannot find any piece of code in your repo addressing this aspect.

Thus, would it be possible for you to share with us (if you have it of course) the code you used to select the best hyper-parameters on Gowalla, Yelp, and Amazon Book? This would be super important to conduct our research further :-)

Thanks in advance!

Best,
Daniele

**Comment by jeongwhanchoi** [comment]
Dear @danielemalitesta 

The repository includes a complete implementation of the BSPM model with configuration options in `parse.py`, but you're right that there's no explicit code for validation-based hyperparameter selection. BSPM follows a paradigm different from learning-based recommendation systems like LightGCN.

BSPM belongs to a family of methods (including GF-CF, PGSP, and FIRE) that don't require conventional training processes. These models are non-parametric approaches that directly compute recommendations through matrix operations and graph-filtering diffusion processes rather than iterative parameter optimization.

In our research, since there is no training phase, we use a framework that proceeds directly to testing without hyperparameter selection from a validation set. This follows the GF-CF framework exactly.

If you examine `model.py`, you'll notice the `train()` method in the `BSPM` class isn't actually training in the traditional sense -- it's a preprocessing step that pre-computes necessary matrices and decompositions.

While we're also interested in more rigorous validation approaches, this code's framework is currently limited to the scope of the GF-CF framework.

Best,
Jeongwhan
